### PR TITLE
[8.x] Fulltext index for PostgreSQL

### PR DESCRIPTION
**Pull request by tpetry**
As promised on Twitter, I want to help with the PostgreSQL support for fulltext search.

But as fulltext searching is no SQL standard, every database has an entirely different implementation. The most significant difference is that PostgreSQL fulltext searching is language-dependent because the stemming algorithms are always only designed for a specific language. Also, adding a fulltext index for multiple columns is different. The vectors for multiple columns need to be combined for the index creation. But when querying the database the exact same syntax for combining the vectors needs to be used for the query or the index will not be used. As there is no query builder method to build these conditions, I restricted the usage to a single column for the moment. With an official way to query the index the restriction could be lifted.

So the following changes had to be made:

- An optional language parameter was added to `Blueprint::fulltext()`
- The PostgreSQL driver will fail if no language was provided
- The PostgreSQL driver will fail when trying to add a fulltext index with multiple columns

### CODE REVIEW

File: src/Illuminate/Database/Schema/Grammars/PostgresGrammar.php [hunk 1]
```diff
@@ -176,6 +177,32 @@ public function compileIndex(Blueprint $blueprint, Fluent $command)
...
+     *
+     * @param  \Illuminate\Database\Schema\Blueprint  $blueprint
+     * @param  \Illuminate\Support\Fluent  $command
+     * @return string
+     *
+     * @throws \RuntimeException
+     */
+    public function compileFulltext(Blueprint $blueprint, Fluent $command)
+    {
+        if (null === $command->language) {
```
Comment (faissaloux) [inline src/Illuminate/Database/Schema/Grammars/PostgresGrammar.php:191]:
```suggestion
        if (is_null($command->language)) {
```

Comment (tpetry) [reply src/Illuminate/Database/Schema/Grammars/PostgresGrammar.php:191]: A specific reason why you suggest replacing a language construct to a function? The behavior will be the same.

Comment (faissaloux) [reply src/Illuminate/Database/Schema/Grammars/PostgresGrammar.php:191]: Consistency

Comment (tpetry) [reply src/Illuminate/Database/Schema/Grammars/PostgresGrammar.php:191]: Both ways are used in laravel. I can‘t say which one is the preferred one 🤷‍♂️

Comment (mfn) [reply src/Illuminate/Database/Schema/Grammars/PostgresGrammar.php:191]: I see Taylor often converting them to the `is_null` variant.

I don't understand why one does this manually and there's no fixer in place doing this automagically 😅

File: CHANGELOG-8.x.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 ## [Unreleased](https://github.com/laravel/framework/compare/v8.74.0...8.x)
 
+### Added
+- Added Fulltext Index support for PostgreSQL
```
Comment (driesvints) [inline CHANGELOG-8.x.md:6]: Please remove this again.

File: src/Illuminate/Database/Schema/Grammars/PostgresGrammar.php (hunk 1 above)
Comment (driesvints) [reply src/Illuminate/Database/Schema/Grammars/PostgresGrammar.php:191]: @mfn what fixer do you recommend?

File: CHANGELOG-8.x.md (hunk 2 above)
Comment (tpetry) [reply CHANGELOG-8.x.md]: Done

File: src/Illuminate/Database/Schema/Grammars/PostgresGrammar.php (hunk 1 above)
Comment (mfn) [reply src/Illuminate/Database/Schema/Grammars/PostgresGrammar.php]: @driesvints none actually 😅

sorry for the noise, was thinking about https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/master/doc/rules/language_construct/is_null.rst but that's the other way around and Style CI can do this too but it's not he CS of the project

**Pull request merged**